src/Refactoring-UI-Tests/RBRenameInstanceVariableDriverTest.class.st

```diff
@@ -22,6 +22,23 @@ RBRenameInstanceVariableDriverTest >> testInvalidName [
 	
 ]
 
+{ #category : 'tests' }
+RBRenameInstanceVariableDriverTest >> testValidNameAndUndefinedVariable [
+	
+	| rbclasses driver |
+	rbclasses := RBClassEnvironment classes: { MyClassARoot}.	
+	driver := RBRenameInstanceVariableDriver new
+				scopes: { rbclasses }
+				variable: #inexistantVariable
+				for: MyClassARoot
+				newName: #inexistantVariable1.
+	"Variable `inexistantVariable` is not defined in `MyClassARoot`"
+	self setUpDriver: driver.
+	
+	self should: [driver runRefactoring] raise: RBRefactoringError
+	
+]
+
 { #category : 'tests' }
```

Comment:
I'm wondering if the driver should raise an exception or just inform the user. 
Now I'm wondering how by construction this could happen because the UI should only present some valid instance variable names to be renamed?

Reply:
The flow right now is that driver raises an exception, and then the exception informs the user.
I think it makes sense to remove even that exception handling from it, and only let it do UI notifications and stop with execution once it's done. Is this what you meant?

Reply:
Yes